poc_mobile_verifier/src/verifier.rs

```diff
@@ -75,24 +76,34 @@ impl VerifierDaemon {
             .verify_epoch(&self.pool, &scheduler.verification_period)
             .await?;
 
-        let mut transaction = self.pool.begin().await?;
+        let transaction = self.pool.begin().await?;
 
         // TODO: switch to a bulk transaction
-        for heartbeat in heartbeats.into_iter() {
-            heartbeat.write(&self.heartbeats_tx).await?;
-            heartbeat.save(&mut transaction).await?;
-        }
-
-        for speedtest in speedtests.clone().without_lapsed().into_iter() {
-            speedtest.write(&self.speedtest_avg_tx).await?;
-        }
-
-        for speedtest in speedtests.into_iter() {
-            speedtest.save(&mut transaction).await?;
-        }
+        let transaction = {
```

Comment:
why the two transaction assignments?

Reply:
The transaction is moved into the `try_fold`, the result of which is the same transaction - so it needs to be re-assigned